packages/tools/src/tools/FreehandROISculptorTool.ts

```diff
@@ -0,0 +1,685 @@
+    );
+    element.addEventListener(
+      Events.MOUSE_CLICK,
+      this._endCallback as EventListener
+    );
+    element.addEventListener(
+      Events.MOUSE_DRAG,
+      this._dragCallback as EventListener
+    );
+  }
+
+  /**
+   * Removes event listeners from the element.
+   * @param element - The viewport element to remove event listeners from.
+   */
+  _deactivateSculpt(element: HTMLDivElement): void {
+    element.removeEventListener(
+      Events.MOUSE_UP,
+      this._endCallback as EventListener
+    );
+    element.removeEventListener(
+      Events.MOUSE_CLICK,
+      this._endCallback as EventListener
+    );
+    element.removeEventListener(
+      Events.MOUSE_DRAG,
+      this._dragCallback as EventListener
+    );
+  }
+
+  _checkSpacing(
+    i: number,
+    points: Array<Types.Point3>,
+    indicesToInsertAfter: Array<number>,
+    maxSpacing: number
+  ): void {
+    const { element } = this._sculptData;
+    const enabledElement = getEnabledElement(element);
+    const { viewport } = enabledElement;
+    const nextHandleIndex = this._getNextHandleIndex(i, points.length);
+
+    const currentCanvasPoint = viewport.worldToCanvas(points[i]);
+    const nextCanvasPoint = viewport.worldToCanvas(points[nextHandleIndex]);
+
+    const distanceToNextHandle = point.distanceToPoint(
+      currentCanvasPoint,
+      nextCanvasPoint
+    );
+
+    if (distanceToNextHandle > maxSpacing) {
+      indicesToInsertAfter.push(i);
+    }
+  }
+
+  _getNextHandleIndex(i: number, length: number): number {
+    if (i === length - 1) {
+      return 0;
+    }
+
+    return i + 1;
+  }
+
+  _getNextHandleIndexBeforeInsert(insertIndex: number, length: number): number {
+    if (insertIndex === length) {
+      return 0;
+    }
+
+    return insertIndex;
+  }
+
+  renderAnnotation(
+    enabledElement: Types.IEnabledElement,
+    svgDrawingHelper: SVGDrawingHelper
+  ): void {
+    const { viewport } = enabledElement;
+    const { element } = viewport;
+
+    const viewportIdsToRender = this._commonData.viewportIdsToRender;
+
+    if (
+      !this._commonData.mouseLocation ||
+      this.mode !== ToolModes.Active ||
+      !viewportIdsToRender.includes(viewport.id)
+    ) {
+      return;
+    }
+
+    const annotations =
+      this._filterInteractableFreehandRoiAnnotationsForElement(element);
+
+    if (!annotations?.length) {
+      return;
+    }
+
+    const styleSpecifier: StyleSpecifier = {
+      toolGroupId: this.toolGroupId,
+      toolName: this.getToolName(),
+      viewportId: enabledElement.viewport.id,
+    };
+
+    let color = getStyleProperty(
+      'color',
+      styleSpecifier,
+      AnnotationStyleStates.Default,
+      this.mode
+    );
+
+    if (this.isActive) {
+      color = getStyleProperty(
+        'color',
+        styleSpecifier,
+        AnnotationStyleStates.Highlighted,
+        this.mode
+      );
+    }
+
+    const circleUID = '0';
+
+    drawCircleSvg(
+      svgDrawingHelper,
+      '',
```

Comment:
Could you use the tool name here - that way you won't accidentally update someone else's drawing.  I know it is annotationUID, but it isn't needed.

Reply:
Updated